src/core/core.legend.js

```diff
@@ -14,12 +14,18 @@ module.exports = function(Chart) {
 
 		// a callback that will handle
 		onClick: function(e, legendItem) {
-			var index = legendItem.datasetIndex;
+			var indices = [];
 			var ci = this.chart;
-			var meta = ci.getDatasetMeta(index);
 
-			// See controller.isDatasetVisible comment
-			meta.hidden = meta.hidden === null? !ci.data.datasets[index].hidden : null;
+			if (!helpers.isArray(legendItem.datasetIndex)) {
+				indices = [legendItem.datasetIndex];
+			}
```

Comment:
In the cade where this is false,  should indices be the array of indexes passed in?